airflow/utils/log/file_task_handler.py

```diff
@@ -18,15 +18,17 @@
 """File logging handler for tasks."""
 import logging
 import os
-from typing import Optional
```

Comment:
I think this should be handled by moving stuff out of models/__init__.py rather by hacking stuff for  TYPE_CHECKING

Reply:
Why not both? This is what the TYPE_CHECKING was designed for.

Reply:
See the #7484 for discussion/proposal of this.

Reply:
_Even_ if importing this one module was quick, it's still an extra import that is not needed at run time. That's a number of directory searches looking for the this one file, and a opening a file, parsing it etc.

Reply:
Not really - it's a hack to hide architectural problems in legacy code.

Reply:
(And the number of directory searches goes up if we use implicit packages.)

Reply:
https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING

Reply:
(I know what the documentation says, it's just my view on what TYPE_CHECKING is often used for)

Reply:
> _Even_ if importing this one module was quick, it's still an extra import that is not needed at run time. That's a number of directory searches looking for the this one file, and a opening a file, parsing it etc.

I believe removing __init__.py  stuff is different (and much better) way of solving the long import time.

When we properly move stuff out of __init__.py's for models (and other places). No single import will be expensive because we will only import what is needed and nothing more. Yes some "types" will be imported needlessly but those will really be types that will be immediately loaded if anything from this function is used and they are probably already imported elsewhere. 

By proper tree-like/lack-of-cycles/no implicit imports structure of the code we will achieve what we want to do now by manual TYPE_CHECKING removal of some stuf - avoid expensive imports where not needed.

Reply:
> (And the number of directory searches goes up if we use implicit namespaces.)

Yeah. I already fully abandoned this idea. I simply stopped being fan of implicit packages. We should not use them. There were many problems in our code that we would have to solve and gain nothing in return, behavior of some code changes when there are implicit packages and there are still many tools/packages (like sphinx but many others) that do not work well with implicit packages yet. It makes little sense to use implicit packages now that we can install providers packages without them.